packages/dom/src/focusable.js

```diff
@@ -33,6 +33,7 @@ function buildSelector( sequential ) {
 		sequential ? '[tabindex]:not([tabindex^="-"])' : '[tabindex]',
 		'a[href]',
 		'button:not([disabled])',
+		'[role="button"][tabindex]',
```

Comment:
Why is this needed, isn't this element caught by the first rule anyway?

Reply:
That was my first assumption, but unfortunately not. It would work fine when you first enter the list as `<div role="button" tabindex="0">` matches the first selector.

However, as you navigate around the list, unfocused elements are given `tabIndex="-1"`, so that results in a `<div role="button" tabindex="-1">` element that doesn't match any of these selectors, meaning you can no longer navigate to them.

I figure adding this selector to the list will make it work like buttons which would match the selector on line 35 when they have a `tabIndex="-1"`.

(though I still think it would be preferable to use a button element)

Reply:
Why unfocused elements are given tabIndex -1 in the first place?

Reply:
That's how roving tab index usually works. The last focused element has tabIndex of 0, and all other focusable elements -1, that way when tabbing back in, focus lands on the last focused element.

Reply:
Mmm, right but the question I have is:

"focusable" is a reusable utility and technically a div with tabIndex -1 is not "focusable" so the "right" thing to do seem to be, to not include these in the results.

Reply:
I don't plan to merge the base PR, so this is a bit of "what if" discussion. However, it seems like elements with button role should be picked up by the focusable package. Note that buttons (or any/most of the rules above) would also have `tabIndex=-1` at some point when using the roving tabindex technique.

Reply:
> Note that buttons (or any/most of the rules above) would also have tabIndex=-1 at some point when using the roving tabindex technique.

This actually seem like a bug to me. A button with tabIndex=-1 is not focusable AFAIK

Reply:
There's some semantics, but I think this is the reason why there's a `tabbable` utility and a `focusable` utility.

From the spec (https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute):
> −1 (or other negative integer values)
Causes the element to be [focusable](https://html.spec.whatwg.org/multipage/interaction.html#focusable), and indicates that the author would prefer the element to be [click focusable](https://html.spec.whatwg.org/multipage/interaction.html#click-focusable) but not [sequentially focusable](https://html.spec.whatwg.org/multipage/interaction.html#sequentially-focusable). The user agent might ignore this preference for click and sequential focusability, e.g., for specific element types according to platform conventions, or for keyboard-only users.

Reply:
Ah, also found this info from the same spec which is relevant:
> Being [focusable](https://html.spec.whatwg.org/multipage/interaction.html#focusable) is a statement about whether an element can be focused programmatically, e.g. via the [focus()](https://html.spec.whatwg.org/multipage/interaction.html#dom-focus) method or [autofocus](https://html.spec.whatwg.org/multipage/interaction.html#attr-fe-autofocus) attribute. In contrast, [sequentially focusable](https://html.spec.whatwg.org/multipage/interaction.html#sequentially-focusable) and [click focusable](https://html.spec.whatwg.org/multipage/interaction.html#click-focusable) govern how the user agent responds to user interaction: respectively, to [sequential focus navigation](https://html.spec.whatwg.org/multipage/interaction.html#sequential-focus-navigation) and as [activation behavior](https://html.spec.whatwg.org/multipage/interaction.html#focus-activation-behavior).

Reply:
Ah makes sense. Thanks for correcting me.